src/agri_ai/core/database_pool.py

```
@@ -0,0 +1,319 @@
+        collection = await self.get_collection(collection_name)
+        result = await collection.delete_one(filter_dict)
+        
+        # キャッシュを無効化
+        self._invalidate_cache(collection_name)
+        
+        return result.deleted_count > 0
+    
+    def _invalidate_cache(self, collection_name: str):
+        """特定のコレクションのキャッシュを無効化"""
+        keys_to_remove = []
+        for key in self.query_cache.keys():
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)